### ISSUE TITLE
Add `prefer-top-level-if-only-type-imports` scenario for `consistent-type-specifier-style`

### ISSUE THREAD

**Issue by FoxxMD** [un-ts/eslint-plugin-import-x#501]
`eslint-plugin-import` has an (unreleased) third option for the [consistent-type-specifier-style](https://github.com/un-ts/eslint-plugin-import-x/blob/master/docs/rules/consistent-type-specifier-style.md) called `prefer-top-level-if-only-type-imports`. From their docs

> `'prefer-top-level-if-only-type-imports'` - enforces that named type-only specifiers must use a top-level, type-only import when all named imports are types; if some are values, then inline markers are allowed. This is useful when you generally prefer inline but **you are using TypeScript verbatimModuleSyntax and want all-type imports omitted by bundlers.**

Emphasis mine. To elaborate on this last bit and why this is so useful:

With the introduction of [Type stripping](https://nodejs.org/api/typescript.html#type-stripping) in node many typescript projects now have the opportunity to run their `.ts` code directly with node rather than needing to compile first, *with some exceptions*.

One of the largest of these exceptions is that node requires explicit [`type` declarations in import statements](https://nodejs.org/api/typescript.html#importing-types-without-type-keyword) in order to erase them correctly. Using the typescript compiler option `verbatimModuleSyntax` makes this easier as typescript produces errors when importing something that is a type but not explicitly declared as such.

*However*, there is one other [discrepancy](https://typescript-eslint.io/rules/consistent-type-imports/#comparison-with-importsnotusedasvalues--verbatimmodulesyntax) between typescript and node WRT this import syntax. Given this import statement

```ts
import { type T } from 'T';
```

* Typescript compilation => emits *nothing* because all imported values were types
* Node runtime => evaluates `import { } from 'T';` and still imports the module

For node, this is mostly an issue for bundlers which "follow" each module imported to determine what to pull in. It can lead to a bundler pulling in a module that should only be used for typing.

The correct way to fix this is to use top-level `type`

```ts
import type { T } from 'T';
```

which fully erases the module import in node.

But, this is only true *if all inline imports are types.* If the import is mixed with types and values its totally fine to use inline types because node ends up seeing the correct value imports after erasing types.

```ts
// mixed import becomes...
import { type Y, T } from 'T';

// this is okay for node!
import { T } from 'T';
```
___

So, I'd like to request that eslint-plugin-import-x support an equivalent of `'prefer-top-level-if-only-type-imports'` for its own consistent-type-specifier-style rule. Thank you!

**Comment by SukkaW** [comment]
I see. But any reason for not using `prefer-top-level`? It enforces:

```ts
import { type Y, T } from 'T';
import { type A } from 'B';
```

into

```ts
import type { Y } from 'T';
import { T } from 'T';

import type A from 'B';
```

which, after stripping, turns into

```ts
//
import { T } from 'T';
//
```

and is OK.

**Comment by FoxxMD** [comment]
You're correct, the end result is the same when using `prefer-top-level` but stylistically it's different.

I like to consolidate all imports for the same module into one statement, when possible, and use something like `import-x/no-duplicates` to enforce that. But with only `prefer-top-level` I'd have to choose between using one rule or the other.
